packages/mdc-shape/_functions.scss

```diff
@@ -143,6 +124,61 @@
   }
 }
 
+//
+// Accepts radius number or list of 2 radius values and returns 4 value list with
+// top-left value, top-right value, bottom-right 0, bottom-left 0.
+// Throws error when length of radius is more than 2.
+//
+// Example:
+//
+// 1. mdc-shape-mask-top-radius((2px, 3px)) => 2px 3px 0 0
+// 2. mdc-shape-mask-top-radius(8px) => 8px 8px 0 0
+//
+@function mdc-shape-mask-top-radius($radius) {
+  @if length($radius) > 2 {
+    @error "Invalid radius: '#{$radius}' component doesn't allow customizing all corners";
+  } @else if length($radius) == 2 {
+    @return nth($radius, 1) nth($radius, 2) 0 0;
+  } @else {
+    @return $radius $radius 0 0;
+  }
+}
+
+//
+// Accepts radius number or list of 2-4 radius values and returns 4 value list with
+// masked corners as mentioned in `$masked-corners`
+//
+// Example:
+//
+// 1. mdc-shape-mask-radius(2px 3px, 1 1 0 0) => 2px 3px 0 0
+// 2. mdc-shape-mask-top-radius(8px, 0 0 1 1) => 0 0 8px 8px
+// 3. mdc-shape-mask-top-radius(4px 4px 4px 4px, 0 1 1 0) => 0 4px 4px 0
+//
+@function mdc-shape-mask-radius($radius, $masked-corners) {
+  @if type-of($radius) == "list" {
+    @if length($radius) > 4 {
+      @error "Invalid radius: '#{$radius}' is more than 4 values";
+    }
+  }
+
+  @if type-of($masked-corners) != "list" and length($masked-corners) != "length" {
```

Comment:
I don't think this condition is quite right... `length($masked-corners)` will never equal `"length"` so this always evaluates to false.

I think this can be simplified to `@if length($masked-corners) != 4` since `length` of something that isn't a list is always `1`.